File: python/day01/main.py
import re

def main():
    input_text = open(0).read()
    print(f'part one: {solve_part_one(input_text)}')
    print(f'part two: {solve_part_two(input_text)}')

def solve_part_one(input_text: str):
    
    total = 0

    try:
    
        for i, line in enumerate(input_text.split('\n')):
            if line == '':
                continue
            
            digits = re.findall('\d', line)
            total += int(digits[0] + digits[-1])
    finally:

        return total


def solve_part_two(input_text: str):
    total = 0

    number_map = (
        'one',
        'two',
        'three',
        'four',
        'five',
        'six',
        'seven',
        'eight',
        'nine',
    )
    
    for i, line in enumerate(input_text.split('\n')):
        if line == '':
            continue
        
        for i, s in enumerate(number_map):
            line = line.replace(s, s[0] + str(i +1) + s[-1])
        
        digits = re.findall('\d', line)

        total += int(digits[0] + digits[-1])

    return total

if __name__ == '__main__':
    raise SystemExit(main())

File: go/cmd/day01/main.go
package main

import (
	"fmt"
	"io"
	"os"
	"regexp"
	"strconv"
	"strings"
)

func solve_part_one(input_text string, replace bool) int {
	total := 0

	subs := []string{"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"}

	for _, line := range strings.Split(input_text, "\n") {
		if replace {
			for i, s := range subs {
				line = strings.ReplaceAll(line, s, fmt.Sprintf("%c%d%c", s[0], i + 1, s[len(s) - 1]))
			}	
		}
		r := regexp.MustCompile("[0-9]")
		digits := r.FindAllString(line, -1)

		if len(digits) == 0 {
			continue
		}

		n, err := strconv.Atoi(fmt.Sprintf("%s%s", digits[0], digits[len(digits) -1]))
		if (err != nil) {
			panic(err)
		}

		total += n
	}

	return total
}

func solve_part_two(input_text string) int {
	return solve_part_one(input_text, true)
}


func main() {
	stdin, err := io.ReadAll(os.Stdin)
	if err != nil {
		panic(err)
	}

	input := string(stdin)


	part_one_result := solve_part_one(input, false)
	part_two_result := solve_part_two(input)
	fmt.Printf("part one: %d\n", part_one_result)
	fmt.Printf("part two: %d\n", part_two_result)
}